stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_connector.py

```diff
@@ -90,7 +96,8 @@ def create_results_connection(self, query, offset, length):
                             ErrorResponder.fill_error(return_obj, response_dict, ['error', 'message'])
                     except KeyError:
                         break
-                return_obj['data'] = return_obj['data'][:length]
+                # slice the cumulative records as per the provided offset and length(limit)
+                return_obj['data'] = return_obj['data'][offset:offset+length]
```

Comment:
though it can be a safe way to return the correct number of results. if you found out already $skip doesn't work correctly than you can keep this slicing.

Reply:
@mdazam1942 •	The $top and $skip param does not work correctly in the sentinel API for some of the queries. For instance, the below example for a $filter query with processes states gives expected results, but the same for file states gives incorrect results. Due to this, there is a need of a programmatic slicing of the cumulative results fetched from the API based on the $top and $offset values as per our current implementation. Kindly check my email for more details on this observation.